client/tests/input_selection/alias_outputs.rs

```diff
@@ -0,0 +1,704 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::str::FromStr;
+
+use iota_client::{
+    api::input_selection::new::{Burn, InputSelection, Requirement},
+    block::{
+        address::Address,
+        output::{AliasId, AliasOutputBuilder, Output},
+        protocol::protocol_parameters,
+    },
+    Error,
+};
+
+use crate::input_selection::{
+    build_alias_output, build_basic_output, build_inputs, unsorted_eq,
+    Build::{Alias, Basic},
+    ALIAS_ID_0, ALIAS_ID_1, ALIAS_ID_2, BECH32_ADDRESS, BECH32_ADDRESS_ALIAS_SENDER, BECH32_ADDRESS_ED25519_SENDER,
+    BECH32_ADDRESS_NFT_SENDER, TOKEN_SUPPLY,
+};
+
+#[test]
+fn input_alias_eq_output_alias() {
+    let protocol_parameters = protocol_parameters();
+    let alias_id_2 = AliasId::from_str(ALIAS_ID_2).unwrap();
+
+    let inputs = build_inputs(vec![Alias(1_000_000, alias_id_2, BECH32_ADDRESS, None, None, None)]);
+    let outputs = vec![build_alias_output(
+        1_000_000,
+        alias_id_2,
+        BECH32_ADDRESS,
+        None,
+        None,
+        None,
+    )];
+
+    let selected = InputSelection::new(inputs.clone(), outputs.clone(), protocol_parameters)
+        .select()
+        .unwrap();
```

Comment:
Shouldn't this fail because input and output has the same state index?

Reply:
Aren't you allowed to do a governance transition for no specific reason ?

Reply:
Oh right, already ignored this case because there is no reason for it :trollface: